### PR TITLE
feat(lint): add a report lint config setting

### PR DESCRIPTION
**Pull request by brenelz**
Builds off this PR to add a "report" setting to deno.json which can be "pretty", "compact", or "json"

https://github.com/denoland/deno/pull/15926

Open to feedback. Struggled a bit with the rust code and probably easier ways to do things

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Looks good, could you update `cli/schemas/config-file.v1.json`?

**Comment by brenelz** [comment]
> Looks good, could you update `cli/schemas/config-file.v1.json`?

updated

File: cli/schemas/config-file.v1.json [hunk 1]
```diff
@@ -249,6 +249,10 @@
               "uniqueItems": true
             }
           }
+        },
+        "report": {
+          "type": "string",
```
Comment (bartlomieju) [inline cli/schemas/config-file.v1.json:254]: I think it might be better to use `enum` here with `default` set to `pretty`

**Review by bartlomieju — CHANGES_REQUESTED** [review]
One small nitpick and it's good to land.

File: cli/schemas/config-file.v1.json (hunk 1 above)
Comment (brenelz) [reply cli/schemas/config-file.v1.json]: should be updated

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks!

**Pull request merged**